File: 0671-second-minimum-node-in-a-binary-tree/0671-second-minimum-node-in-a-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def findSecondMinimumValue(self, root: Optional[TreeNode]) -> int:
        myset = set();
        def dfs(node):
            if not node: return;
            myset.add(node.val);
            # if node.left:
            dfs(node.left);
            # if node.right:
            dfs(node.right);

        dfs(root);
        if len(myset) < 2:
            return -1;
        return sorted(myset)[1];

File: 0671-second-minimum-node-in-a-binary-tree/0671-second-minimum-node-in-a-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int findSecondMinimumValue(TreeNode root) {
        Set<Integer> myset = new HashSet<Integer>();
        dfs(root, myset);
        if (myset.size() < 2){
            return -1;

        }

        List<Integer> sortedList = new ArrayList<>(myset);
        Collections.sort(sortedList);
        return sortedList.get(1);

    }

    private void dfs(TreeNode node, Set<Integer> myset) {
        if (node == null) {
            return;
        }
        myset.add(node.val);
        dfs(node.left, myset);
        dfs(node.right, myset);
    }
    

}
